File: 백준/Silver/11048. 이동하기/이동하기.cc
#include <stdio.h>
int arr[1004][1004], dp[1004][1004], i, j, n, m;
int max(int a, int b) {
	return (a > b) ? a : b;
}
int main() {
	scanf("%d %d", &n, &m);
	for (i = 1; i <= n; i++)
		for (j = 1; j <= m; j++)
			scanf("%d", &arr[i][j]);
	for (i = 1; i <= n; i++) {
		for (j = 1; j <= m; j++) {
			dp[i][j] = arr[i][j] + max(max(dp[i][j - 1], dp[i - 1][j]), dp[i - 1][j - 1]);
		}
	}
	printf("%d", dp[n][m]);
	return 0;
}

File: 백준/Silver/11048. 이동하기/이동하기.py
import sys


def input(): return sys.stdin.readline().rstrip()


N, M = map(int, input().split())

arr = [list(map(int, input().split())) for _ in range(N)]
dp = [[0] * (M + 1) for _ in range(N + 1)]

for i in range(N):
    for j in range(M):
        dp[i][j] = arr[i][j] + \
            max(dp[i][j - 1], dp[i - 1][j], dp[i - 1][j - 1])

print(dp[N - 1][M - 1])